### ISSUE TITLE
onChange doesn't reflect changes when using with React Router 4's component prop

### ISSUE THREAD

**Issue by binarybaba** [paypal/downshift#441]
* `downshift` version: 1.31.10
* `node` version: 9.6.1
* `yarn` version: 1.5.1
* `react router` version: 4.2.2 

Relevant code or config

```
// A scenario like this doesn't work
<Route component={props => (
          <DownshiftComponent
                onChange={props.onChange}
          />
     )
 }>
// however, this does
// A scenario like this doesn't work
<Route render={props => (
          <DownshiftComponent
                onChange={props.onChange}
          />
     )
 }>

```

What happened:
onChange doesn't reflect the selectedItem inside the input box when using with React router 4's component property on `<Route/>`.

Reproduction repository: https://codesandbox.io/s/xojn9mkz54

Problem description:
When using React router 4 combined with passing onChange handler as props to the Downshift component, the onChange gets fired but the selectedItem isn't updated in the input box.

Suggested solution:
I used the render prop instead of the component prop in my `<Route/>` to get it working after like 5 days of banging my head against the wall not knowing what was going wrong. But I'm posting this as an issue along with a (hacky?) workaround for those who encounter the same problem. I don't know if this is an issue or not but well.

**Comment by austintackaberry** [comment]
I don't think this is a hacky solution. For React Router v4, the component prop accepts a component, whereas the render prop accepts a function.

**Comment by binarybaba** [comment]
Welp. I just started with React and react Router 4 so I hope if beginners stumble upon this problem not knowing what's going on, this could help.. Can someone close this issue?
